tests/mock_crust/churn.rs

```diff
@@ -210,9 +259,26 @@ fn churn() {
         // expected_gets.send_and_expect(data_id, auth_s0, auth_g0, &nodes, min_group_size);
         // expected_gets.send_and_expect(data_id, auth_s0, auth_n0, &nodes, min_group_size);
 
-        poll_and_resend(&mut nodes, &mut []);
+        // Test messages from a client to a group and a section...
+        let data_id1 = DataIdentifier::Immutable(rng.gen());
+        let auth_g2 = Authority::ClientManager(rng.gen());
+        let auth_s1 = Authority::Section(section_name);
+        expected_gets.client_send_and_expect(data_id1, client_auth, auth_g2, &clients, &mut nodes);
+        expected_gets.client_send_and_expect(data_id1, client_auth, auth_s1, &clients, &mut nodes);
+        // ... and from group to the client
+        let data = gen_immutable_data(&mut rng, 8);
+        let auth_nae = Authority::NaeManager(*data.name());
+        expected_gets.send_to_client(data.clone(),
+                                     auth_nae,
+                                     client_auth,
+                                     &mut nodes,
+                                     min_group_size);
 
+        let _ = poll_all(&mut nodes, &mut clients);
```

Comment:
Is `poll_and_resend` not necessary anymore? That's not really plausible, if we're dropping nodes, I think?